### ISSUE TITLE
Fails to collect state and phone data

### ISSUE THREAD

**Issue by hesco** [progressivetech/net.ourpowerbase.remoteform#44]
Thank you for this extension.  It seems to mostly be meeting my requirements on the sites I am working on at the moment.  

My initial installation of this extension wants to collect data for a profile seeking first_name, last_name, email, state (US only) and phone number.  I am successfully collecting first and last name and email.  The displayed form fails to display a field for the phone number.  It does ask for state and for whatever reason only displays US states in the drop down box.  (I cannot remember if this is because I configured that somewhere or not).  But the state fails to make it into the database on form submission.  I suspect I might have to somehow pass a country code to make the state data relevant, but have not figured out a way to do so.  

Any guidance you might offer would be appreciated.

**Comment by jmcclelland** [comment]
I recently had to [patch core](https://github.com/civicrm/civicrm-core/pull/20309) to get counties to properly show up. Sadly, api v3  is a bit rickety and not very well supported now that api v4 is on the scene. I suspect we may need another core patch to get phone numbers working. 

Try running this API call via the command line or the API explorer: `cv api Profile.getfields api_action=submit profile_id=YOURPROFILEID` - and see if the phone number shows up. If it does show up, then this is probably a remoteform bug. If it does not show up, then it's a core api v3 bug.

In the long run, this extension will either need to switch to api v4, or somehow work with afform.

**Comment by hesco** [comment]
I presume this indicates a bug with RemoteForm, then.  

```
        "phone-primary-1": {                                                                                                                                                                    
            "name": "phone",
            "type": 2,
            "title": "Phone-Phone (Primary)",
            "description": "Complete phone number.",
            "maxlength": 32,
            "size": 20,
            "import": true,
            "where": "civicrm_phone.phone",
            "headerPattern": "/phone/i",
            "dataPattern": "/^[\\d\\(\\)\\-\\.\\s]+$/",
            "export": true,
            "table_name": "civicrm_phone",
            "entity": "phone",
            "bao": "CRM_Core_BAO_Phone",
            "localizable": 0,
            "html": {
                "type": "Text",
                "label": "Phone",
                "maxlength": 32,
                "size": 20
            },
            "add": "1.1",
            "is_core_field": true,
            "api.required": null,
            "help_pre": null,
            "help_post": null,
            "weight": "6",
            "api.aliases": [
                "phone-Primary-1"
            ]                                                                                                                                                                                   │
        },
```

**Comment by hesco** [comment]
The failure to exhibit the optional phone field in the remote form is unfortunate, but far more concerning is the failure to record in the database the data offered by the user in response to the visible and required country_id and state_province_id fields.  Any thoughts on the matter?  Our form has now been published and is collecting (incomplete) data, and the response is growing.  Would be great to resolve this before there is much more incomplete data which must be manually updated.

**Comment by jmcclelland** [comment]
For the phone field - it looks like you have specifed "primary" as the location in the profile. But that's not enough to know where location type to use. Can you try changing the profile to specify an actual location (home, work, etc)? 

As for states - try adding the country field to the form and I think it should work properly.

**Comment by hesco** [comment]
I found that choosing an option other than `primary` for each of the contact fields was enough to get them stored in the database.  I also found that I can exclude the country, and the form knows to default to the U.S. as its default when presenting the state option selector.  Again, I assume that I set that default somewhere, possibly as that of the `Default Organization Address` on this page: `/civicrm/admin/domain`.
